core/src/main/java/dk/alexandra/fresco/framework/network/Network.java

```diff
@@ -63,4 +72,51 @@ default void sendToAll(byte[] data) {
     }
   }
 
+  /**
+   * Creates a map of party IDs to network address (uses localhost and supplied ports) information.
+   *
+   * @param n the number of parties
+   * @param ports the ports to be used
+   * @return a map of party IDs to network address information
+   */
+  static Map<Integer, NetworkConfiguration> getNetworkConfigurations(int n,
+      List<Integer> ports) {
+    Map<Integer, NetworkConfiguration> confs = new HashMap<>(n);
+    Map<Integer, Party> partyMap = new HashMap<>();
+    int id = 1;
+    for (int port : ports) {
+      partyMap.put(id, new Party(id, "localhost", port));
+      id++;
+    }
+    for (int i = 0; i < n; i++) {
+      confs.put(i + 1, new NetworkConfigurationImpl(i + 1, partyMap));
+    }
+    return confs;
+  }
+
+  /**
+   * Finds {@code portsRequired} free ports and returns their port numbers. <p>NOTE: two subsequent
+   * calls to this method can return overlapping sets of free ports (same with parallel calls).</p>
+   *
+   * @param portsRequired number of free ports required
+   * @return list of port numbers of free ports
+   */
+  static List<Integer> getFreePorts(int portsRequired) {
```

Comment:
See above, these methods should not be on the interface. Particularly this is only useful for tests (or possibly local demos).

Reply:
Moved